test/main/promise.js

```diff
@@ -152,3 +157,31 @@ QUnit.test( "fulfilled Promise with non-Promise async assertion", function( asse
 	// Adds 1 assertion
 	return createMockPromise( assert );
 } );
+
+QUnit.module( "Proise-aware return values with rejections", {
+	afterEach: function( assert ) {
+		assert.test.pushFailure = this.pushFailure;
+	}
+} );
+
+QUnit.test( "rejected Promise with undefined value", function( assert ) {
+	this.pushFailure = assert.test.pushFailure;
+	assert.test.pushFailure = function( message ) {
+		assert.equal(
```

Comment:
Should this be strictEqual? (I tend to favor strictEqual for string compares, so ignore this if repo conventions say use equal for string comparisons)